server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MultipleDatabaseChangeRuntimeFlowableTask.java

```diff
@@ -0,0 +1,232 @@
+
+    private Integer batchId;
+
+    private Integer batchNumber;
+
+    @Autowired
+    private FlowInstanceService flowInstanceService;
+
+    @Autowired
+    private FlowableAdaptor flowableAdaptor;
+
+    @Autowired
+    private ServiceTaskInstanceRepository serviceTaskInstanceRepository;
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+
+    @Override
+    public boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return false;
+    }
+
+    @Override
+    protected Void start(Long taskId, TaskService taskService, DelegateExecution execution)
+            throws InterruptedException {
+        TaskContextHolder.trace(authenticationFacade.currentUser().getId(), taskId);
+        try {
+            log.info("multiple database task start, taskId={}", taskId);
+            TaskEntity detail = taskService.detail(taskId);
+            MultipleDatabaseChangeParameters multipleDatabaseChangeParameters = JsonUtils.fromJson(
+                    detail.getParametersJson(), MultipleDatabaseChangeParameters.class);
+            Integer value = multipleDatabaseChangeParameters.getBatchId();
+            if (value == null) {
+                this.batchId = 0;
+            } else {
+                this.batchId = value;
+            }
+            multipleDatabaseChangeParameters.setBatchId(this.batchId + 1);
+            detail.setParametersJson(JsonUtils.toJson(multipleDatabaseChangeParameters));
+            taskService.updateParametersJson(detail);
+            this.batchNumber = multipleDatabaseChangeParameters.getOrderedDatabaseIds().size();
+            List<Long> batchDatabaseIds =
+                    multipleDatabaseChangeParameters.getOrderedDatabaseIds().get(this.batchId);
+            List<Long> flowInstanceIds = new ArrayList<>();
+            for (Long batchDatabaseId : batchDatabaseIds) {
+                CreateFlowInstanceReq createFlowInstanceReq = new CreateFlowInstanceReq();
+                createFlowInstanceReq.setDatabaseId(batchDatabaseId);
+                createFlowInstanceReq.setTaskType(TaskType.ASYNC);
+                createFlowInstanceReq.setExecutionStrategy(FlowTaskExecutionStrategy.AUTO);
+                createFlowInstanceReq.setParentFlowInstanceId(FlowTaskUtil.getFlowInstanceId(execution));
+                createFlowInstanceReq.setParameters(multipleDatabaseChangeParameters
+                        .convertIntoDatabaseChangeParameters(multipleDatabaseChangeParameters));
+                List<FlowInstanceDetailResp> individualFlowInstance = flowInstanceService.createIndividualFlowInstance(
+                        createFlowInstanceReq);
+                flowInstanceIds.add(individualFlowInstance.get(0).getId());
+            }
+
+            long originalTime = System.currentTimeMillis();
+
+            boolean flag = true;
```

Comment:
what is the flag means, may refine naming

Reply:
'flag' has been changed to 'flagForTaskSucceed',databases in the same batch are checked after finished, and any database failure will set 'flagForTaskSucceed' to false